rust/arrow/src/datatypes.rs

```diff
@@ -1903,9 +1929,20 @@ mod tests {
 
     #[test]
     fn serde_struct_type() {
+        let kv_array = [("k".to_string(), "v".to_string())];
```

Comment:
We should serialise the metadata to a JSON struct in the form of:

```json
{
  "metadata": [
    {"key": "k", "value": "v"}
  ]
}
```

There are integration tests that we need to enable, I'll find the relevant file in the repository.

Reply:
Fixed in `impl From<&Field> for ArrowJsonField`, thanks!

Reply:
@nevi-me I just saw the comment "The JSON can either be an Object or an Array of Objects" in from [ARROW-8883: [Rust] [Integration] Enable more tests](https://github.com/apache/arrow/commit/18b92818e7e12c6516511abe33b9e9ff78573571#diff-16170ebfd75c15cd0ef92a3e4f35dba353edcb11352101c363fc35cf3729267d)

That's interesting! But I haven't seen the `Object` example from testing/data/arrow-ipc-stream/integration/1.0.0-littleendian/generated_custom_metadata.json.gz. So would you please give some pointers?

Reply:
I can't remember where I encountered it, might be in the `0.14.1` test files. A quick way of seeing where the object example comes from, is to comment out that portion of the code, and run the integration tests.

Reply:
> I can't remember where I encountered it, might be in the `0.14.1` test files. A quick way of seeing where the object example comes from, is to comment out that portion of the code, and run the integration tests.

@nevi-me  After comment out, datatypes::tests::schema_json failed, but the test data are not from `testing` module, instead are manually constructed. Files with "metadata" in the file name can only be found in these directories: 1.0.0-bigendian and 1.0.0-littleendian/. The Object format for Schema metadata came from https://github.com/apache/arrow/pull/5907 by @grundprinzip

Anyway, I think it's no harm to support parse Field metadata from JSON Object(Map), just as what Schema did.
What do you think?

Reply:
Sorry, forgot to reply earlier. Yeah, there's no harm in keeping the behaviour as is in this PR